### PR TITLE
AP_TECS: fix broken logging bitmask enable

### PR DESCRIPTION
**Pull request by Hwurzburg**

### CODE REVIEW

File: libraries/AP_TECS/AP_TECS.cpp [hunk 1]
```diff
@@ -5,6 +5,8 @@
 #include <AP_Logger/AP_Logger.h>
 #include <AP_Landing/AP_Landing.h>
 
+#define MASK_LOG_TECS                   (1<<11)
```
Comment (IamPete1) [inline libraries/AP_TECS/AP_TECS.cpp:8]: Should be passed in the constructor. Or at least leave a big comment in the plane explaining that it is re-defined here.

Comment (magicrub) [reply libraries/AP_TECS/AP_TECS.cpp:8]: agreed, this mask like this is a total hack.

Comment (Hwurzburg) [reply libraries/AP_TECS/AP_TECS.cpp:8]: being a poor programmer, I can only go by the patterns established in the code:

 like here: https://github.com/ArduPilot/ardupilot/blob/1c78baac72659a70cbdad055030242e10c9c9194/libraries/AP_Baro/AP_Baro.cpp#L951

and  here https://github.com/ArduPilot/ardupilot/blob/1c78baac72659a70cbdad055030242e10c9c9194/libraries/AP_Generator/AP_Generator_RichenPower.cpp#L312

and here https://github.com/ArduPilot/ardupilot/blob/1c78baac72659a70cbdad055030242e10c9c9194/libraries/AP_Compass/AP_Compass.cpp#L1633

and here https://github.com/ArduPilot/ardupilot/blob/1c78baac72659a70cbdad055030242e10c9c9194/libraries/AP_Generator/AP_Generator_IE_2400.cpp#L188

**Review by hendjoshsr71 — COMMENTED** [review]
I approve, but I helped so...

File: libraries/AP_TECS/AP_TECS.cpp (hunk 1 above)
Comment (Hwurzburg) [reply libraries/AP_TECS/AP_TECS.cpp]: changed

File: libraries/AP_TECS/AP_TECS.h [hunk 2]
```diff
@@ -161,7 +162,10 @@ class AP_TECS {
 
     // reference to const AP_Landing to access it's params
     const AP_Landing &_landing;
-    
+
+    // Logging  bitmask
+    uint32_t _log_bitmask;
```
Comment (IamPete1) [inline libraries/AP_TECS/AP_TECS.h:167]: this could be const.

**Review by IamPete1 — APPROVED** [review]
Change is correct, I guess just a question of the change in behavior but I don't think losing TECS messages because we now honor the bit-mask is really a big issue. I wonder how long it has been like this..

**Comment by IamPete1** [comment]
Looks like its been broken ever since TECS logging was moved into TECS and out of plane. At least 5 years.

File: libraries/AP_TECS/AP_TECS.h (hunk 2 above)
Comment (Hwurzburg) [reply libraries/AP_TECS/AP_TECS.h]: done

**Review by IamPete1 — APPROVED** [review]

**Review by tridge — APPROVED** [review]

**Pull request merged**